### PR TITLE
Add macOS support

### PR DESCRIPTION
**Pull request by wshanks**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by wshanks** [comment]
This is passing now! It includes a patch for the undeclared `_SC_PHYS_PAGES` issue. I mentioned the patch upstream [here](https://github.com/Qiskit/qiskit-aer/issues/1923). I will wait for feedback on that. When something viable is accepted upstream, I will backport it and release a macOS build (i.e. I don't plan to release a build with a patch that upstream does not like).

**Review by basnijholt — APPROVED** [review]
Awesome, let's go! :tada:

I plan to work on the MacOS ARM64 build after this.

File: recipe/meta.yaml
Comment (basnijholt) [inline recipe/meta.yaml:39]: Should probably revert this though.

Comment (wshanks) [reply recipe/meta.yaml:39]: Yes, there are some changes in here to limit the CI just to save some trees. I don't have a mac so I did all my testing in the CI here. I will take them out before merging.

**Comment by wshanks** [comment]
@basnijholt Since you are interested in the mac build, I will clean this up for merging. I had been holding out trying to get a response from upstream so that I wouldn't have to keep a patch in the recipe.

**Comment by wshanks** [comment]
@leofang I just noticed there were no osx jobs :slightly_smiling_face:

File: recipe/meta.yaml
Comment (leofang) [inline recipe/meta.yaml:31]: nitpick: maybe we don't need to patch for other OSs?
```suggestion
    - 0001-Use-Apple-specific-API-to-determine-system-memory-on.patch  # [osx]
```

Comment (leofang) [reply recipe/meta.yaml:31]: (Either we keep the status quo, and bump the build number, or we limit the patch applicability and keep the same build number.)

Comment (wshanks) [reply recipe/meta.yaml:31]: Oh, yes, I forgot about the build number after resetting this branch back to `main`. We can limit the patch to osx. Since the CI is pretty far along, I will let it finish to see if there are any other issues before updating the branch.

Comment (leofang) [reply recipe/meta.yaml:31]: Yup let's let it finish, no rush

**Review by leofang — APPROVED** [review]

**Comment by leofang** [comment]
The one CI failure can be ignored, it's a glitch that should go away when merging into main.

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:38]: Given this PR adds macOS, maybe this comment should be dropped as it is no longer relevant?

```suggestion
```

Comment (wshanks) [reply recipe/meta.yaml:38]: Oops! I'll take that comment out with the next update. My macOS patch was accepted upstream closing the linked issue there, so with the next update we can take out the Windows and macOS patches and that comment.

Comment (jakirkham) [reply recipe/meta.yaml:38]: No worries. It happens. Opened issue ( https://github.com/conda-forge/qiskit-aer-feedstock/issues/24 ) to track

**Pull request merged**
